magda-web-client/src/UI/MarkdownViewer.js

```diff
@@ -1,36 +1,29 @@
 import React from 'react';
-import marked from 'marked';
+import MarkdownIt from 'markdown-it';
 import './MarkdownViewer.css';
 
-const LIST = ['blockquote',
-       'br',
-       'code',
-       'codespan',
-       'del',
-       'em',
-       'heading',
-       'hr',
-       'html',
-       'image',
-       'link',
-       'list',
-       'listitem',
-       'paragraph',
-       'strong',
-       'table',
-       'tablecell',
-       'tablerow'];
 
-let renderer = new marked.Renderer();
-LIST.forEach(l=>{
- renderer[l] = function (text) {
-   return text;
- };
-});
+class MarkdownViewer extends React.Component  {
+  constructor(props) {
+   super(props);
+   this.state = {isCollapsed: true};
+ }
 
-function MarkdownViewer(props) {
-    let markdown = {__html: marked(props.markdown)};
-    return <div className='markdown' dangerouslySetInnerHTML={markdown}/>
+ componentDidMount(){
+   const numberOfChildrenFromMarkdownAsHtml = this.markdown.children.length;
```

Comment:
This should probably happen more often than just the initial mount, right?  Like if the markdown changes and the content length increases, a user of `MarkdownViewer` would probably expect `updateContentLength` to be called again.